### PR TITLE
new app insights ingredient

### PR DESCRIPTION
**Pull request by bdschaap**
Draft.  Feedback appreciated.  

The ARM template includes Smart Detection alerts which the portal auto-creates if not included.  I'm including them to allow customization as well as to disable email alerts.  Otherwise, just about everyone with an an account receives an email.  One of the alerts also relies on a specific naming convention of Azure automatically creates a duplicate alert.  I don't want to spend too much time on these though since Microsoft is retiring Smart Detection alerts on June 30th, 2019 and plans to provide a migration path.

### CODE REVIEW

**Comment by JasonPagel** [comment]
You should move the dependency of @azbake/core from the dependencies section to the devDependencies in your package.json

**Comment by csperbeck** [comment]
Why should Core be moved to dev dependencies? Is core not required to compile the code and is in a direct import statement, and NPM doesn't pull devDependencies by default, neither does lerna. The `--hoist` we are using does, but default does not. Just curious @JasonPagel

**Comment by whilke** [comment]
We use peerDependencies for ingredients so that only one version of core is loaded. peerDeps are never installed by npm/lerna, and the recommended setup is to add them as devDeps so that lerna hoist can make sure they get installed correctly for local debug. System installs core directly for the real runtime execution.

**Comment by csperbeck** [comment]
Cool beans. Just curious as I have a couple of PRs incoming. Thanks @whilke

**Comment by csperbeck** [comment]
Also, rebasing the branch onto master, it fails the `npm clean:build` ... 
```
lerna ERR! npm run compile exited 2 in '@azbake/ingredient-app-insights'
lerna ERR! npm run compile stdout:

> @azbake/ingredient-app-insights@0.0.1 compile C:\Users\csperbeck\source\repos\azure-bake\ingredient\ingredient-app-insights
> tsc -p ./src

../../node_modules/@azbake/arm-helper/dist/arm-helper.d.ts(1,62): error TS7016: Could not find a declaration file for module '@azbake/core'. 'C:/Users/csperbeck/source/repos/azure-bake/node_modules/@azbake/core/dist/index.js' implicitly has an 'any' type.
  Try `npm install @types/azbake__core` if it exists or add a new declaration (.d.ts) file containing `declare module '@azbake/core';`
```

**Comment by csperbeck** [comment]
Did you test the ingredient? Hook it up to your personal azure environment, etc?

**Comment by csperbeck** [comment]
Maybe we should include a `/test` folder in each ingredient? Something to key a functional test off of?

**Comment by bdschaap** [comment]
> Did you test the ingredient? Hook it up to your personal azure environment, etc?

Yes I was able to use the ingredient to deploy an App Insights resource to Azure

**Comment by bdschaap** [comment]
> Maybe we should include a `/test` folder in each ingredient? Something to key a functional test off of?

Yes I like the idea.  Happy to add my test yaml to a /test folder if you would like me to.

File: ingredient/ingredient-app-insights/package-lock.json [hunk 1]
```diff
@@ -0,0 +1,954 @@
+{
+  "name": "@azbake/ingredient-app-insights",
+  "version": "0.0.1",
+  "lockfileVersion": 1,
+  "requires": true,
+  "dependencies": {
+<<<<<<< HEAD
```
Comment (whilke) [inline ingredient/ingredient-app-insights/package-lock.json]: This needs to be fixed

Comment (whilke) [reply ingredient/ingredient-app-insights/package-lock.json]: @bdschaap Resolve this lock file and I can get this merged, released today.

**Comment by whilke** [comment]
@csperbeck I saw you committed several items to master. Make sure to do that through a PR, and all commits to master need to follow conventional commit message format so that we don't break changelogs.

**Comment by HomegrownDev** [comment]
Yes, sir. That was the Azure Devops commits for the build system for the
build.yaml. Will direct it though PRs for future improvements.

Charlie Sperbeck

**Comment by whilke** [comment]
build systems can commit to master, we just need to make sure they use the right commit messages, which should normally be a chore: type commit.

File: ingredient/ingredient-app-insights/package-lock.json (hunk 1 above)
Comment (bdschaap) [reply ingredient/ingredient-app-insights/package-lock.json]: fixed (regenerated) the lock file

**Pull request merged**